currency/storage.go

```diff
@@ -48,142 +78,132 @@ func (s *Storage) SetDefaults() {
-			}
-
-		case "CurrencyLayer":
-			if overrides.FxCurrencyLayer || settings.ForexProviders[i].Enabled {
-				settings.ForexProviders[i].Enabled = true
-				fxSettings = append(fxSettings,
-					base.Settings(settings.ForexProviders[i]))
-			}
+		enabled := (settings.ForexProviders[i].Name == "CurrencyConverter" && overrides.CurrencyConverter) ||
+			(settings.ForexProviders[i].Name == "CurrencyLayer" && overrides.CurrencyLayer) ||
+			(settings.ForexProviders[i].Name == "Fixer" && overrides.Fixer) ||
+			(settings.ForexProviders[i].Name == "OpenExchangeRates" && overrides.OpenExchangeRates) ||
+			(settings.ForexProviders[i].Name == "ExchangeRates" && overrides.ExchangeRates) ||
+			(settings.ForexProviders[i].Name == "ExchangeRateHost" && overrides.ExchangeRateHost)
+
+		if !enabled {
+			continue
+		}
 
-		case "Fixer":
-			if overrides.FxFixer || settings.ForexProviders[i].Enabled {
-				settings.ForexProviders[i].Enabled = true
-				fxSettings = append(fxSettings,
-					base.Settings(settings.ForexProviders[i]))
+		if settings.ForexProviders[i].Name != DefaultForexProviderExchangeRatesAPI {
+			if settings.ForexProviders[i].APIKey == "" || settings.ForexProviders[i].APIKey == "Key" {
+				log.Warnf(log.Currency, "%s forex provider API key not set, disabling. Please set this in your config.json file\n",
+					settings.ForexProviders[i].Name)
+				continue
 			}
 
-		case "OpenExchangeRates":
-			if overrides.FxOpenExchangeRates ||
-				settings.ForexProviders[i].Enabled {
-				settings.ForexProviders[i].Enabled = true
-				fxSettings = append(fxSettings,
-					base.Settings(settings.ForexProviders[i]))
+			if settings.ForexProviders[i].APIKeyLvl == -1 && settings.ForexProviders[i].Name != "ExchangeRates" {
+				log.Warnf(log.Currency, "%s APIKey level not set, functionality is limited. Please review this in your config.json file\n",
+					settings.ForexProviders[i].Name)
 			}
+		}
 
-		case "ExchangeRates":
-			// TODO ADD OVERRIDE
-			if settings.ForexProviders[i].Enabled {
-				settings.ForexProviders[i].Enabled = true
-				fxSettings = append(fxSettings,
-					base.Settings(settings.ForexProviders[i]))
+		if settings.ForexProviders[i].PrimaryProvider {
+			if primaryProvider {
+				log.Warnf(log.Currency, "%s disabling primary provider, multiple primarys found. Please review providers in your config.json file\n",
+					settings.ForexProviders[i].Name)
+				settings.ForexProviders[i].PrimaryProvider = false
+			} else {
+				primaryProvider = true
 			}
+		}
+		fxSettings = append(fxSettings, base.Settings(settings.ForexProviders[i]))
+	}
 
-		case "ExchangeRateHost":
-			if overrides.FxExchangeRateHost || settings.ForexProviders[i].Enabled {
-				settings.ForexProviders[i].Enabled = true
-				fxSettings = append(fxSettings,
-					base.Settings(settings.ForexProviders[i]))
+	if len(fxSettings) == 0 {
+		log.Warnln(log.Currency, "No foreign exchange providers enabled, setting default provider....")
```

Comment:
_four_ periods???????????? 😠 
```suggestion
		log.Warnln(log.Currency, "No foreign exchange providers enabled, setting default provider...")
```